torchvision/prototype/transforms/functional/_meta.py

```diff
@@ -57,41 +58,86 @@ def convert_bounding_box_format(
 
 
+_rgb_to_grayscale_tensor = _FT.rgb_to_grayscale
+
+
 def convert_image_color_space_tensor(
     image: torch.Tensor, old_color_space: ColorSpace, new_color_space: ColorSpace
 ) -> torch.Tensor:
     if new_color_space == old_color_space:
         return image.clone()
 
-    if old_color_space == ColorSpace.GRAYSCALE:
-        image = _grayscale_to_rgb_tensor(image)
-
-    if new_color_space == ColorSpace.GRAYSCALE:
-        image = _FT.rgb_to_grayscale(image)
-
-    return image
-
-
-def _grayscale_to_rgb_pil(grayscale: PIL.Image.Image) -> PIL.Image.Image:
-    return grayscale.convert("RGB")
-
-
-def convert_image_color_space_pil(
-    image: PIL.Image.Image, old_color_space: ColorSpace, new_color_space: ColorSpace
-) -> PIL.Image.Image:
-    if new_color_space == old_color_space:
-        return image.copy()
-
-    if old_color_space == ColorSpace.GRAYSCALE:
-        image = _grayscale_to_rgb_pil(image)
-
-    if new_color_space == ColorSpace.GRAYSCALE:
-        image = _FP.to_grayscale(image)
-
-    return image
+    if old_color_space == ColorSpace.OTHER or new_color_space == ColorSpace.OTHER:
+        raise RuntimeError(f"Conversion to or from {ColorSpace.OTHER} is not supported.")
+
+    if old_color_space == ColorSpace.GRAYSCALE and new_color_space == ColorSpace.GRAYSCALE_ALPHA:
+        return _add_alpha(image)
+    elif old_color_space == ColorSpace.GRAYSCALE and new_color_space == ColorSpace.RGB:
+        return _grayscale_to_rgb_tensor(image)
+    elif old_color_space == ColorSpace.GRAYSCALE and new_color_space == ColorSpace.RGBA:
+        return _add_alpha(_grayscale_to_rgb_tensor(image))
+    elif old_color_space == ColorSpace.GRAYSCALE_ALPHA and new_color_space == ColorSpace.GRAYSCALE:
+        return _strip_alpha(image)
+    elif old_color_space == ColorSpace.GRAYSCALE_ALPHA and new_color_space == ColorSpace.RGB:
+        return _grayscale_to_rgb_tensor(_strip_alpha(image))
+    elif old_color_space == ColorSpace.GRAYSCALE_ALPHA and new_color_space == ColorSpace.RGBA:
+        image, alpha = _split_alpha(image)
+        return _add_alpha(_grayscale_to_rgb_tensor(image), alpha)
+    elif old_color_space == ColorSpace.RGB and new_color_space == ColorSpace.GRAYSCALE:
+        return _rgb_to_grayscale_tensor(image)
+    elif old_color_space == ColorSpace.RGB and new_color_space == ColorSpace.GRAYSCALE_ALPHA:
+        return _add_alpha(_rgb_to_grayscale_tensor(image))
+    elif old_color_space == ColorSpace.RGB and new_color_space == ColorSpace.RGBA:
+        return _add_alpha(image)
+    elif old_color_space == ColorSpace.RGBA and new_color_space == ColorSpace.GRAYSCALE:
+        return _rgb_to_grayscale_tensor(_strip_alpha(image))
+    elif old_color_space == ColorSpace.RGBA and new_color_space == ColorSpace.GRAYSCALE_ALPHA:
+        image, alpha = _split_alpha(image)
+        return _add_alpha(_rgb_to_grayscale_tensor(image), alpha)
+    else:  # old_color_space == ColorSpace.RGBA and new_color_space == ColorSpace.RGB:
```

Comment:
Nit: Perhaps leave it as an `elif` and throw an error if anything else. This way it will continue behaving if we add something new on the future.